src/transformers/models/mpt/modeling_flax_mpt.py

```diff
@@ -0,0 +1,698 @@
+        self,
+        hidden_states,
+        position_bias: jnp.ndarray,
+        attention_mask: Optional[Tuple[jnp.ndarray]] = None,
+        past_key_value: Optional[Tuple[jnp.ndarray]] = None,
+    ):
+        batch_size, seq_length = hidden_states.shape[:2]
+
+        mixed_qkv = self.Wqkv(hidden_states)
+        query_states, key_states, value_states = jnp.split(mixed_qkv, 3, axis=-1)
+        query_states = query_states.reshape((batch_size, seq_length, self.n_heads, self.head_dim)).transpose(
+            (0, 2, 1, 3)
+        )
+        key_states = key_states.reshape((batch_size, seq_length, self.n_heads, self.head_dim)).transpose((0, 2, 1, 3))
+        value_states = value_states.reshape((batch_size, seq_length, self.n_heads, self.head_dim)).transpose(
+            (0, 2, 1, 3)
+        )
+
+        if past_key_value is not None:
+            if past_key_value[0].shape[2] > 0:
+                key_states = jnp.concatenate([past_key_value[0], key_states], axis=2)
+                value_states = jnp.concatenate([past_key_value[1], value_states], axis=2)
+            past_key_value = (key_states, value_states)
+        else:
+            past_key_value = (key_states, value_states)
+
+        attention_scores = jnp.matmul(query_states, key_states.transpose((0, 1, 3, 2))) * self.softmax_scale
+
+        query_length = seq_length if past_key_value is None else seq_length + past_key_value[0].shape[2]
+
+        if position_bias is not None:
+            if position_bias.ndim != 3:
+                raise ValueError(f"Expecting position_bias shape to be 3 dimensions, got {position_bias.ndim}")
+            key_length = key_states.shape[2]
+
+            position_bias_query_index = max(0, position_bias.shape[1] - query_length)
+            position_bias_key_index = max(0, position_bias.shape[2] - key_length)
+
+            position_bias = position_bias[:, position_bias_query_index:, position_bias_key_index:]
+
+            attention_scores = attention_scores + position_bias
+
+        if attention_mask is not None:
+            attention_scores = jax.lax.select(
+                attention_mask, jnp.full_like(attention_scores, -jnp.inf), attention_scores
+            )
+
+        attn_weights = nn.softmax(attention_scores, axis=-1).astype(value_states.dtype)
+        attn_weights = nn.dropout(attn_weights, rate=self.attn_dropout_p, deterministic=not self.training)
+
+        context_states = jnp.matmul(attn_weights, value_states)
+        context_states = context_states.transpose((0, 2, 1, 3)).reshape((batch_size, seq_length, -1))
+        attn_output = nn.Dense(features=self.hidden_dim, kernel_init=nn.initializers.xavier_uniform())(context_states)
+
+        return attn_output, attn_weights, past_key_value
+
+
+class FlaxMptMLP(nn.Module):
+    hidden_size: int
+    attn_pdrop: float
+
+    def setup(self):
+        self.up_proj = nn.Dense(
+            features=4 * self.hidden_size, kernel_init=nn.initializers.xavier_uniform(), use_bias=False
```

Comment:
Again could you refer to GPT Neo to see how we initialise these linear layers please?